plugins/org.python.pydev.parser/src/org/python/pydev/parser/grammar310/TreeBuilder310.java

```diff
@@ -611,7 +611,11 @@ public final SimpleNode onCloseNode(SimpleNode n, int arity) throws Exception {
                         guard = (exprType) securePopNullable(exprType.class);
                     }
                     patternType pattern = (patternType) securePop(patternType.class);
-                    return new match_caseType(pattern, guard, suite.body);
+                    match_caseType matchCase = new match_caseType(pattern, guard, suite.body);
+                    if (guard != null) {
+                        guard.parent = matchCase;
```

Comment:
I don't think that the `parent` should be set here (I guess you did it for the workaround in the `visitCompare` where `!(node.parent instanceof match_caseType))` is done, but that has a different solution and this shouldn't be needed).